src/osd/PrimaryLogPG.cc

```diff
@@ -7072,9 +7072,7 @@ int PrimaryLogPG::do_osd_ops(OpContext *ctx, vector<OSDOp>& ops)
 	    p.second.set_flag(chunk_info_t::FLAG_MISSING);
```

Comment:
The commit message doesn't seem to match this change.

Reply:
Also, does setting object_info_t::size actually fix the problem you mention in the commit message?  It seems like you need to check the chunk_map to check whether the object is evicted during recovery.